src/output_writers.jl

```diff
@@ -270,126 +270,217 @@ end
-    elseif fw.naming_scheme == :file_number
-        file_num = Int(iteration / fw.frequency)
-        fw.prefix * name * lpad(file_num, fw.padding, "0") * ext(fw)
-    else
-        throw(ArgumentError("Invalid naming scheme: $(fw.naming_scheme)"))
-    end
+Examples
+========
+
+julia> netcdf_spatial_dimensions(model.velocities.u)
+("xF", "yC", "zC")
+
+julia> netcdf_spatial_dimensions(model.tracers.T)
+("xC", "yC", "zC")
+"""
+function netcdf_spatial_dimensions(field::Field{LX, LY, LZ}) where {LX, LY, LZ}
+    xdim(LX), ydim(LY), zdim(LZ)
 end
 
-function write_output(model::Model, fw::NetCDFOutputWriter)
-    fields = Dict(
+xdim(::Type{Face}) = "xF"
+xdim(::Type{Cell}) = "xC"
+ydim(::Type{Face}) = "yF"
+ydim(::Type{Cell}) = "yC"
+zdim(::Type{Face}) = "zF"
+zdim(::Type{Cell}) = "zC"
+
+# This function converts and integer to a range, and nothing to a Colon
+get_slice(n::Integer) = n:n
+get_slice(n::UnitRange) = n
+get_slice(n::Nothing) = Colon()
+
+"""
+    write_grid(model; filename="./grid.nc", mode="c", 
+               compression=0, attrib=Dict(), slice_kw...)
+
+Writes a grid.nc file that contains all the dimensions of the domain.
+
+Keyword arguments
+=================
+
+    - `filename::String`  : File name to be saved under
+    - `mode::String`      : Netcdf file is opened in either clobber ("c") or append ("a") mode (Default: "c" )
+    - `compression::Int`  : Defines the compression level of data (0-9, default 0)
+    - `attrib::Dict`      : Attributes (default: Dict())
+"""
+function write_grid(model; filename="./grid.nc", mode="c",
+                    compression=0, attrib=Dict(), slice_kw...)
+    dimensions = Dict(
         "xC" => collect(model.grid.xC),
         "yC" => collect(model.grid.yC),
         "zC" => collect(model.grid.zC),
-        "xF" => collect(model.grid.xF),
-        "yF" => collect(model.grid.yF),
-        "zF" => collect(model.grid.zF),
-        "u" => Array(parentdata(model.velocities.u)),
-        "v" => Array(parentdata(model.velocities.v)),
-        "w" => Array(parentdata(model.velocities.w)),
-        "T" => Array(parentdata(model.tracers.T)),
-        "S" => Array(parentdata(model.tracers.S))
+        "xF" => collect(model.grid.xF)[1:end-1],
+        "yF" => collect(model.grid.yF)[1:end-1],
+        "zF" => collect(model.grid.zF)[1:end-1]
+    )
+    dim_attrib = Dict(
+        "xC" => ["longname" => "Locations of the cell centers in the x-direction.", "units" => "m"],
+        "yC" => ["longname" => "Locations of the cell centers in the y-direction.", "units" => "m"],
+        "zC" => ["longname" => "Locations of the cell centers in the z-direction.", "units" => "m"],
+        "xF" => ["longname" => "Locations of the cell faces in the x-direction.",   "units" => "m"],
+        "yF" => ["longname" => "Locations of the cell faces in the y-direction.",   "units" => "m"],
+        "zF" => ["longname" => "Locations of the cell faces in the z-direction.",   "units" => "m"]
```

Comment:
Hmmm, `"units" => "m"` is true by default, but will be wrong for `NonDimensionalModel`. Don't think we have to worry about that in this PR though.

Reply:
It is a bit confusing. Technically speaking there are no assumptions about units in the code. The user is free to pick their parameters with the desired units and as long as everything the user does is consistent, the results will be correct.

Reply:
Would be difficult to add a units property to all fields and grid axes? That way we could just call it.

Reply:
This may not be difficult. Alternatively, we can look into using `Unitful`, which might be more natural / less hack-y. Is it necessary to have units? I think it could be best to leave the units unspecified, and create an issue about adding units to the model, since this is a larger problem than should be addressed in this PR probably.